api_test.go

```diff
@@ -0,0 +1,230 @@
+		ctx := context.Background()
+		index := "rkci"
+		field := "f"
+
+		_, err := m0.API.CreateIndex(ctx, index, pilosa.IndexOptions{Keys: false})
+		if err != nil {
+			t.Fatalf("creating index: %v", err)
+		}
+		_, err = m0.API.CreateField(ctx, index, field, pilosa.OptFieldTypeSet(pilosa.DefaultCacheType, 100), pilosa.OptFieldKeys())
+		if err != nil {
+			t.Fatalf("creating field: %v", err)
+		}
+
+		rowKey := "rowkey"
+
+		// Generate some keyed records.
+		rowKeys := []string{rowKey, rowKey, rowKey}
+		colIDs := []uint64{1, 2, pilosa.ShardWidth + 1}
+		timestamps := []int64{0, 0, 0}
+
+		// Import data with keys to the coordinator (node0) and verify that it gets
+		// translated and forwarded to the owner of shard 0 (node1; because of moddHasher)
+		req := &pilosa.ImportRequest{
+			Index:      index,
+			Field:      field,
+			Shard:      0,
+			RowKeys:    rowKeys,
+			ColumnIDs:  colIDs,
+			Timestamps: timestamps,
+		}
+		if err := m0.API.Import(ctx, req); err != nil {
+			t.Fatal(err)
+		}
+
+		pql := fmt.Sprintf("Row(%s=%s)", field, rowKey)
+
+		// Query node0.
+		if res, err := m0.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if columns := res.Results[0].(*pilosa.Row).Columns(); !reflect.DeepEqual(columns, colIDs) {
+			t.Fatalf("unexpected column ids: %+v", columns)
+		}
+
+		// Query node1.
+		if res, err := m1.API.Query(ctx, &pilosa.QueryRequest{Index: index, Query: pql}); err != nil {
+			t.Fatal(err)
+		} else if columns := res.Results[0].(*pilosa.Row).Columns(); !reflect.DeepEqual(columns, colIDs) {
+			t.Fatalf("unexpected column ids: %+v", columns)
+		}
+	})
+}
+func TestAPI_ImportValue(t *testing.T) {
+	c := test.MustRunCluster(t, 2,
+		[]server.CommandOption{
+			server.OptCommandServerOptions(
+				pilosa.OptServerNodeID("node0"),
+				pilosa.OptServerClusterHasher(&moddHasher{}),
+			)},
+		[]server.CommandOption{
+			server.OptCommandServerOptions(
+				pilosa.OptServerNodeID("node1"),
+				pilosa.OptServerClusterHasher(&moddHasher{}),
+			)},
+	)
+	defer c.Close()
+
+	m0 := c[0]
+	m1 := c[1]
+
+	t.Run("ValColumnKey", func(t *testing.T) {
+		ctx := context.Background()
+		index := "valck"
+		field := "f"
+
+		_, err := m0.API.CreateIndex(ctx, index, pilosa.IndexOptions{Keys: true})
+		if err != nil {
+			t.Fatalf("creating index: %v", err)
+		}
+		_, err = m0.API.CreateField(ctx, index, field, pilosa.OptFieldTypeInt(0, 100))
+		if err != nil {
+			t.Fatalf("creating field: %v", err)
+		}
+
+		// Generate some keyed records.
+		values := []int64{}
+		colKeys := []string{}
+		for i := 1; i <= 10; i++ {
+			values = append(values, int64(i))
+			colKeys = append(colKeys, fmt.Sprintf("col%d", i))
+		}
+
+		// Import data with keys to the coordinator (node0) and verify that it gets
+		// translated and forwarded to the owner of shard 0 (node1; because of moddHasher)
+		req := &pilosa.ImportValueRequest{
+			Index:      index,
+			Field:      field,
+			Shard:      0,
```

Comment:
I think it would be better not to explicitly set this here since (a) it will be 0 anyway and (b) it should be ignored since we're using column keys. Seeing it set explicitly was confusing to me (maybe I'm missing something though).